src/thunderbots/software/test/world/game_state.cpp

```diff
@@ -0,0 +1,366 @@
+#include <gtest/gtest.h>
+
+#include "test/test_util/test_util.h"
+#include "util/refbox_constants.h"
+
+// tuple of start state, update state, end state, our_restart, restart reason
+typedef std::tuple<RefboxGameState, RefboxGameState, RefboxGameState, bool,
+                   GameState::RestartReason>
+    StateTransitionTestParams;
+
+class GameStateTransitionTest : public ::testing::TestWithParam<StateTransitionTestParams>
+{
+};
+
+TEST_P(GameStateTransitionTest, test_state_transitions)
+{
+    RefboxGameState start_state             = std::get<0>(GetParam());
+    RefboxGameState update_state            = std::get<1>(GetParam());
+    RefboxGameState end_state               = std::get<2>(GetParam());
+    bool our_restart                        = std::get<3>(GetParam());
+    GameState::RestartReason restart_reason = std::get<4>(GetParam());
+
+    GameState game_state;
+    game_state.updateRefboxGameState(start_state);
+    game_state.updateRefboxGameState(update_state);
+    EXPECT_EQ(game_state.getRefboxGameState(), end_state);
+    EXPECT_EQ(game_state.isOurRestart(), our_restart);
+    EXPECT_EQ(game_state.getRestartReason(), restart_reason);
+}
+
+#define STATE_TRANSITION_PARAMS(start, update, end, our_restart, restart_reason)         \
```

Comment:
Quick javadoc comment for the arguments here would be good (even though it's not a function).